dev_guide/configmaps.adoc

```diff
@@ -0,0 +1,616 @@
+----
+====
+
+[[real-world-example-configuring-redis]]
+## Real World Example: Configuring Redis
+
+Let's take a look at a real-world example: configuring redis using ConfigMap.  Say we want to inject
+redis with the recommendation configuration for using redis as a cache.  The redis config file
+should contain:
+
+====
+----
+maxmemory 2mb
+maxmemory-policy allkeys-lru
+----
+====
+
+Say that we have such a file in `example-files/redis`; we can use the following command to create a
+ConfigMap instance with it:
+
+====
+----
+$ oc create configmap example-redis-config --from-file=example-files/redis/redis-config
+
+$ oc get configmap redis-config -o yaml
+----
+====
+
+====
+[source:json]
+----
+{
+    "kind": "ConfigMap",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "example-redis-config",
+        "namespace": "default",
+        "selfLink": "/api/v1/namespaces/default/configmaps/example-redis-config",
+        "uid": "07fd0419-d97b-11e5-b443-68f728db1985",
+        "resourceVersion": "15",
+        "creationTimestamp": "2016-02-22T15:43:34Z"
+    },
+    "data": {
+        "redis-config": "maxmemory 2mb\nmaxmemory-policy allkeys-lru\n"
+    }
+}
+----
+====
+
+Now, let's create a pod that uses this config:
+
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: redis
+spec:
+  containers:
+  - name: redis
+    image: kubernetes/redis:v1
+    env:
+    - name: MASTER
+      value: "true"
+    ports:
+    - containerPort: 6379
+    resources:
+      limits:
+        cpu: "0.1"
+    volumeMounts:
+    - mountPath: /redis-master-data
+      name: data
+    - mountPath: /redis-master
+      name: config
+  volumes:
+    - name: data
+      emptyDir: {}
+    - name: config
+      configMap:
+        name: example-redis-config
+        items:
+        - key: redis-config
+          path: redis.conf
+----
+====
+
+Notice that this pod has a ConfigMap volume that places the `redis-config` key of the
+`example-redis-config` ConfigMap into a file called `redis.conf`.  This volume is mounted into the
+`/redis-master` directory in the redis container, placing our config file at
+`/redis-master/redis.conf`, which is where the image looks for the redis config file for the master.
+
+====
+----
+$ kubectl create -f docs/user-guide/configmap/redis/redis-pod.yaml
+----
+====
+
+If we `oc exec` into this pod and run the `redis-cli` tool, we can check that our config was
+applied correctly:
+
+====
+----
+$ oc exec -it redis redis-cli
+127.0.0.1:6379> CONFIG GET maxmemory
+1) "maxmemory"
+2) "2097152"
+127.0.0.1:6379> CONFIG GET maxmemory-policy
+1) "maxmemory-policy"
+2) "allkeys-lru"
+----
+====
+
+## Restrictions
+
+ConfigMaps must be created before they are consumed in pods.  Controllers may be written to tolerate
+missing configuration data; consult individual components configured via ConfigMap on a case-by-case
+basis.
+
+ConfigMaps reside in a namespace.   They can only be referenced by pods in the same namespace.
```

Comment:
Can we s/namespace/project/g here? Trying to err on the side of referring to projects when we can, but mention "namespace" when we can't avoid it (i.e., `--all-namespaces`).